Content.Server/GameObjects/Components/Mobs/Body/Limb.cs

```diff
@@ -0,0 +1,322 @@
+                organ.Initialize(owner, body);
+                Organs.Add(organ);
+            }
+            if (reflectionManager.TryParseEnumReference(TargetKey, out var @enum))
+            {
+                AttackTarget = (AttackTargetDef)@enum;
+            }
+            foreach(var _addTargetKey in _addTargetKeys)
+            {
+                if (reflectionManager.TryParseEnumReference(_addTargetKey.Key, out var @anotherEnum))
+                {
+                    _addTargets.Add(new AdditionalTarget((AttackTargetDef)@anotherEnum, _addTargetKey.Value));
+                }
+            }
+        }
+
+        public LimbRender Render() //TODO
+        {
+            var color = new Color(128, 128, 128);
+            switch (State)
+                {
+                case LimbState.Healthy:
+                    color = new Color(0, 255, 0);
+                    break;
+                case LimbState.InjuredLightly:
+                    color = new Color(255, 255, 0);
+                    break;
+                case LimbState.Injured:
+                    color = new Color(255, 165, 0);
+                    break;
+                case LimbState.InjuredSeverely:
+                    color = new Color(255, 0, 0);
+                    break;
+                case LimbState.Missing:
+                    color = new Color(128, 128, 128);
+                    break;
+                }
+            return new LimbRender(TexturePath, color);
+
+        }
+
+        public void HandleGib()
+        {
+            if(directOrganDamage) //don't spawn both arm and hand, leg and foot
+            {
+                return;
+            }
+            if (_seed.Prob(0.7f))
+            {
+                foreach (var organ in Organs)
+                {
+                    organ.HandleGib();
+                }
+            } 
+            else 
+            {
+                SpawnPrototypeEntity();
+            }
+        }
+
+        public void HandleDecapitation(bool spawn)
+        {
+            State = LimbState.Missing;
+            foreach (var organ in Organs)
+            {
+                organ.State = OrganState.Dead;
+            }
+            foreach (var child in Children)
+            {
+                child.HandleDecapitation(childOrganDamage);
+            }
+            if (spawn)
+            {
+                SpawnPrototypeEntity();
+            }
+        }
+
+        private void Dispose()
+        {
+            Children = null;
+            Organs = null;
+        }
+
+        public void SpawnPrototypeEntity()
+        {
+            //TODO
+            if (!string.IsNullOrWhiteSpace(PrototypeEntity))
+            {
+                Owner.EntityManager.TrySpawnEntityAt(PrototypeEntity, Owner.Transform.GridPosition, out var entity);
+            }
+        }
+
+        public void HandleDamage(int damage) //TODO: test prob numbers, and add targetting!
+        {
+            if (State == LimbState.Missing)
```

Comment:
Shouldn't a state of missing be signified by the limb... not existing, not even a `Limb` instance?